c2corg_ui/static/js/documentediting.js

```diff
@@ -595,6 +594,39 @@ app.DocumentEditingController.prototype.updateMaxSteps = function(waypointType)
 
 
 /**
+ * Check what properties are missing and tell the user.
+ * @param {Object} doc
+ * @export
+ */
+app.DocumentEditingController.prototype.checkMissingProps = function(doc) {
+  var type = doc.type ? app.utils.getDoctype(doc.type) : this.modelName_ + 's'; // missing 's' in modelName if creating a new document
+  var fields = app.constants.REQUIRED_FIELDS[type];
+  var missing = {};
+  var error = [];
+
+  for (var i = 0; i < fields.length; i++) {
+    if (fields[i] === 'title' || fields[i] === 'lang') {
+      missing[fields[i]] = (!doc['locales'] || doc['locales'][0][fields[i]].length === 0) ? true : false;
+    } else if (fields[i] === 'activities') {
+      missing[fields[i]] = (!doc['activities'] || doc['activities'].length === 0) ? true : false;
+    } else if (fields[i] === 'routes') {
+      missing[fields[i]] = (!doc['associations'] || doc['associations']['routes'].length === 0) ? true : false;
+    } else if (fields[i] === 'latitude' || fields[i] === 'longitude') {
+      missing[fields[i]] = (!doc['lonlat'] || doc['lonlat'][fields[i]] == (null || undefined)) ? true : false;
+    } else {
+      missing[fields[i]] = doc[fields[i]] == (null || undefined) ? true : false;
```

Comment:
``? true : false`` is a redundant notation. It should rather be something like that:
```
missing[fields[i]] = (doc[fields[i]] == null || undefined);
```